rust/agama-server/src/software/web.rs

```diff
@@ -49,14 +49,33 @@ pub struct SoftwareConfig {
 /// It emits the Event::ProductChanged and Event::PatternsChanged events.
 ///
 /// * `connection`: D-Bus connection to listen for events.
-pub async fn software_stream(
+pub async fn software_streams(
     dbus: zbus::Connection,
-) -> Result<Pin<Box<dyn Stream<Item = Event> + Send>>, Error> {
-    let stream = StreamExt::merge(
-        product_changed_stream(dbus.clone()).await?,
-        patterns_changed_stream(dbus.clone()).await?,
-    );
-    Ok(Box::pin(stream))
+) -> Result<Streams, Error> {
+    let result: Streams = vec![
+        (
+            "patterns_changed",
+            Box::pin(patterns_changed_stream(dbus.clone()).await?),
+        ),
+        (
+            "product_changed",
+            Box::pin(product_changed_stream(dbus.clone()).await?),
+        ),
+        (
+            "registration_requirement_changed",
+            Box::pin(registration_requirement_changed_stream(dbus.clone()).await?),
```

Comment:
If we move the registration requirement to the product itself (after all, that's part of the product, right?), we do not need this signal.

Reply:
that is right, but value is computed in registration interface and not product one (on dbus side).

Reply:
JFTR, we agreed on postponing this change.